xds/src/main/java/io/grpc/xds/ClusterManagerLoadBalancer.java

```diff
@@ -75,4 +104,58 @@ public String toString() {
       }
     };
   }
+
+  private class ClusterManagerLbState extends ChildLbState {
+    @Nullable
+    ScheduledHandle deletionTimer;
+
+    public ClusterManagerLbState(Object key, LoadBalancerProvider policyProvider,
+        Object childConfig, SubchannelPicker initialPicker) {
+      super(key, policyProvider, childConfig, initialPicker);
+    }
+
+    @Override
+    protected void shutdown() {
+      if (deletionTimer != null && deletionTimer.isPending()) {
+        deletionTimer.cancel();
+      }
+      super.shutdown();
+    }
+
+    @Override
+    protected void reactivate(LoadBalancerProvider policyProvider) {
+      if (deletionTimer != null && deletionTimer.isPending()) {
+        deletionTimer.cancel();
+        logger.log(XdsLogLevel.DEBUG, "Child balancer {0} reactivated", getKey());
+      }
+
+      super.reactivate(policyProvider);
+    }
+
+    @Override
+    protected void deactivate() {
+      if (isDeactivated()) {
+        return;
+      }
+
+      class DeletionTask implements Runnable {
```

Comment:
Some words on why this deletion cannot happen now and has to be scheduled for later?